src/java.base/share/classes/java/lang/foreign/MemorySession.java

```diff
@@ -167,11 +188,21 @@
      */
     void close();
 
+    /**
+     * Returns a non-closeable view of this memory session. The returned session is the same session as this
```

Comment:
By "same" do you mean sames as in `==` rather than `equals`?

Might be clearer to say:
"
If this session is non-closeable then this session is returned.
Otherwise, ...
In the latter case the returned session will be equal to this session as specified by MemorySession.equals.
"
?